### PR TITLE
Update Installation instructions after latest releases

### PR DESCRIPTION
**Pull request by Yadunund**
This PR updates the README here with information to install Open-RMF binaries and notes for developers & maintainers following the recent maintenance overhaul https://github.com/open-rmf/rmf/issues/357

### CODE REVIEW

File: docs/Development-and-Release.md [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+### Release
+* Open-RMF binaries for new ROS 2 distributions will be available along with the new release of ROS 2.
+* Roughly monthly, patch releases for packages with changes will be available in source and binary forms. All releases will be announced [here](https://github.com/open-rmf/rmf/releases)
+  * Source releases are in the form of a [tagged rmf.repos](https://github.com/open-rmf/rmf/blob/release-humble-230606/rmf.repos) file. The least `rmf.repos` file in `<distro>-release` branches contain the latest tags for all the repos for that `<distro>`. The `release-<distro>-YYMMDD` tags are useful for checking out packages to particular release versions.
+  * Binary packages may be updated by running `sudo apt update && sudo apt upgrade -y`.
+
+
+### Additional information for maintainers
+* We release into ROS 2 Rolling from the `main` source branch and from `<distro>` branches into supported ROS 2 distributions.
+* Reference this [dashbaord](https://osrf.github.io/osr_dashboard/?distribution=rmf-rolling) to get an overview of Open-RMF repositories with any new commits since the last release.
```
Comment (aaronchongth) [inline docs/Development-and-Release.md:22]:
```suggestion
* Reference this [dashboard](https://osrf.github.io/osr_dashboard/?distribution=rmf-rolling) to get an overview of Open-RMF repositories with any new commits since the last release.
```

File: README.md [hunk 2]
```diff
@@ -9,151 +9,151 @@ The Open-RMF platform for multi-fleet robot management.
...
-vcs import src < rmf.repos
+sudo apt update && sudo apt install ros-<distro>-rmf-dev
 ```
 
-Ensure all ROS 2 prerequisites are fulfilled,
+**Note:** This will install all necessary debian packages to run Open-RMF except for those in [rmf_demos](https://github.com/open-rmf/rmf_demos) for reasons described in this [issue](https://github.com/open-rmf/rmf_demos/issues/166).
 
-you can subsutitute your distro name for `<your ros distro>`
+To build `rmf_demos` from source, first determine the version of `rmf_demos` that corresponds to the latest binary release for your distro.
+For example, if you installed the `humble` binaries of Open-RMF, the version of `rmf_demos` for this release can be found in the `rmf.repos` file in the `humble-release` branch of this repository, ie, [here](https://github.com/open-rmf/rmf/blob/591b0b9f34a021d81522a0af6f42ffcb5fc70293/rmf.repos#L5)
```
Comment (aaronchongth) [inline README.md:78]:
```suggestion
For example, if you installed the `humble` binaries of Open-RMF, the version of `rmf_demos` for this release can be found in the `rmf.repos` file in the `humble-release` branch of this repository, ie, [here](https://github.com/open-rmf/rmf/blob/humble-release/rmf.repos#L5).
```

File: README.md [hunk 3]
```diff
@@ -9,151 +9,151 @@ The Open-RMF platform for multi-fleet robot management.
...
 
-> NOTE: Due to newer changes in the source build, there might be conflicts and compilation errors with older header files installed by the binaries. Please remove the binary installations before building from source, using `sudo apt remove ros-humble-rmf*`.
-
-Compiling on `Ubuntu 22.04`:
-
-##### Install clang
+### Building from source
+First follow the instruction in the [Setup section](#setup).
 
+Due to newer changes in the source code, there might be conflicts and compilation errors with older header files installed by the binaries.Hence before proceeding, ensure to uninstall all Open-RMF binaries for the ROS 2 distro that will be sourced for the build.
```
Comment (aaronchongth) [inline README.md:96]:
```suggestion
Due to newer changes in the source code, there might be conflicts and compilation errors with older header files installed by the binaries. Hence before proceeding, make sure to uninstall all Open-RMF binaries for the ROS 2 distro that will be sourced for the build.
```

File: README.md [hunk 4]
```diff
@@ -9,151 +9,151 @@ The Open-RMF platform for multi-fleet robot management.
...
-Ensure all ROS 2 prerequisites are fulfilled,
+**Note:** This will install all necessary debian packages to run Open-RMF except for those in [rmf_demos](https://github.com/open-rmf/rmf_demos) for reasons described in this [issue](https://github.com/open-rmf/rmf_demos/issues/166).
 
-you can subsutitute your distro name for `<your ros distro>`
+To build `rmf_demos` from source, first determine the version of `rmf_demos` that corresponds to the latest binary release for your distro.
+For example, if you installed the `humble` binaries of Open-RMF, the version of `rmf_demos` for this release can be found in the `rmf.repos` file in the `humble-release` branch of this repository, ie, [here](https://github.com/open-rmf/rmf/blob/591b0b9f34a021d81522a0af6f42ffcb5fc70293/rmf.repos#L5)
 
-Example:
-for `humble`
+Next create a workspace and build `rmf_demos` from source.
```
Comment (aaronchongth) [inline README.md:80]:
```suggestion
Next create a ROS 2 workspace and build `rmf_demos` from source.
```

File: README.md [hunk 5]
```diff
@@ -9,151 +9,151 @@ The Open-RMF platform for multi-fleet robot management.
...
 
-[//]: # (A good way to install the `rmf` set of packages in one go is to install the one of the main [RMF Demos]&#40;https://github.com/open-rmf/rmf_demos&#41; packages. This will pull all the rest of the Open-RMF packages as a dependency. The core of Open-RMF demos is contained on the `rmf_demos` package. However, if you want to install it with simulation support, you should install the `rmf_demos_gz` or `rmf_demos_gz_classic` package which come with Gazebo or Gazebo Classic support respectively. As an example, to install the ROS 2 Humble release with Gazebo support package, you would run:)
+export CXX=clang++
+export CC=clang
+colcon build --mixin release lld
+```
 
-[//]: # ()
-[//]: # (```bash)
+> NOTE: The first time the build occurs, many simulation models will be downloaded from Ignition Fuel to populate the scene when the simulation is run. As a result, the first build can take a very long time depending on the server load and your Internet connection (for example, 60 minutes). To build without downloading models append `--cmake-args -DNO_DOWNLOAD_MODELS=On` to the `colcon build`
```
Comment (aaronchongth) [inline README.md:148]:
```suggestion
> NOTE: The first time the build occurs, many simulation models will be downloaded from Gazebo Fuel to populate the scene when the simulation is run. As a result, the first build can take a very long time depending on the server load and your internet connection (may be up to an hour). To build without downloading models append `--cmake-args -DNO_DOWNLOAD_MODELS=On` to the `colcon build`
```

File: docs/Development-and-Release.md [hunk 6]
```diff
@@ -0,0 +1,32 @@
+## Development and Release
+
+The development, versioning and release of Open-RMF follows that of the [ROS 2 project](https://docs.ros.org/en/rolling/).
+
+### Development and versioning
+* The primary development branch is `main` on all repositories. This is the equivalent to `rolling` in ROS 2. Any `rolling` branches in Open-RMF repos should be ignored as they exist purely for legacy reasons.
+* Repositories also have ROS 2 distro branches, eg., `humble`, containing versions of the packages distributed as a binary for that distro.
+* All new features and bug fixes must target `main`. The changes may be backported  via cherry-picks to one or more distro branches provided there are no API/ABI breaks.
```
Comment (aaronchongth) [inline docs/Development-and-Release.md:8]:
```suggestion
* All new features and bug fixes must target `main`. The changes may be backported via cherry-picks to one or more distro branches provided there are no API/ABI breaks.
```

File: docs/Development-and-Release.md [hunk 7]
```diff
@@ -0,0 +1,32 @@
...
+* The version of packages on `main` must always be higher than on distro branches (atleast 1 minor version higher) and newer distro versions must be higher than older ones (again, 1 minor version higher).
```
Comment (aaronchongth) [inline docs/Development-and-Release.md:9]:
```suggestion
* The version of packages on `main` must always be higher than on distro branches (at least 1 minor version higher) and newer distro versions must be higher than older ones (again, 1 minor version higher).
```

File: docs/Development-and-Release.md [hunk 8]
```diff
@@ -0,0 +1,32 @@
...
+* Reference this [dashbaord](https://osrf.github.io/osr_dashboard/?distribution=rmf-rolling) to get an overview of Open-RMF repositories with any new commits since the last release.
+* Git `tags` are the most important thing to get right about making a release. A `tag` is associated with a commit and `bloom` will look for new tags in the source branch when making a release. If a commit in a PR was tagged and that PR was squash merged in, the `tag` will not be present in the source branch after the merge.
+* The release repositories for Open-RMF are found in the [ros2-gbp](https://github.com/ros2-gbp) Github organization.
+* Follow [these](https://docs.ros.org/en/rolling/How-To-Guides/Releasing/Releasing-a-Package.html) instructions to release a package and use `cakin_generate_changelogs` to update the changelog files, `catkin_prepare_release` to bump package versions, and `bloom-release` to perform the release for a specified distro. If you do not have access to push the changes directly to the branch, please open a PR. But if the push failed, please ensure to delete any `tags` that were created both locally and upstream before opening the PR.
```
Comment (aaronchongth) [inline docs/Development-and-Release.md:25]:
```suggestion
* Follow [these](https://docs.ros.org/en/rolling/How-To-Guides/Releasing/Releasing-a-Package.html) instructions to release a package and use `catkin_generate_changelogs` to update the changelog files, `catkin_prepare_release` to bump package versions, and `bloom-release` to perform the release for a specified distro. If you do not have access to push the changes directly to the branch, please open a PR. But if the push failed, please ensure to delete any `tags` that were created both locally and upstream before opening the PR.
```

File: docs/Development-and-Release.md [hunk 9]
```diff
@@ -0,0 +1,32 @@
...
+* Please be mindful of the following when making a release for a repository:
+  * Ensure sure that the [tracks.yaml](https://github.com/ros2-gbp/rmf_task-release/blob/master/tracks.yaml) for a repository has a track configured for the ROS 2 distro for which you are making the release.
```
Comment (aaronchongth) [inline docs/Development-and-Release.md:27]:
```suggestion
  * Ensure that the [tracks.yaml](https://github.com/ros2-gbp/rmf_task-release/blob/master/tracks.yaml) for a repository has a track configured for the ROS 2 distro for which you are making the release.
```

**Review by aaronchongth — CHANGES_REQUESTED** [review]
Just some typos, otherwise LGTM!

File: README.md (hunk 2 above)
Comment (Yadunund) [reply README.md:78]: Hmm I'd prefer the permalink as the command below references `2.0.3`. The version might be different if we always retrieve L5 from humle-release/rmf.repos.

Comment (aaronchongth) [reply README.md:78]: Gotcha, then perhaps we should rephrase it to

```suggestion
For example, if you installed the `humble` binaries of Open-RMF, the version of `rmf_demos` for this release can be found in the `rmf.repos` file in the `humble-release` branch of this repository, for example [here](https://github.com/open-rmf/rmf/blob/591b0b9f34a021d81522a0af6f42ffcb5fc70293/rmf.repos#L5).
```

since we are not providing the exact branch we are mentioning.

Comment (Yadunund) [reply README.md]: Much better

**Review by aaronchongth — APPROVED** [review]
LGTM!
